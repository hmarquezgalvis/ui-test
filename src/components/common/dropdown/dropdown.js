import React, { useState } from 'react';
import styled from 'styled-components';

export const DropDown = ({ value,  options = [], onSelected = () => {} }) => {
  const [isActive, setIsActive] = useState(false);

  const toggling = () => setIsActive(!isActive);

  const changeSelected = (item) => {
    setIsActive(false);
    onSelected(item);
  };

  return (
    <DropDownContainer>
      <DropDownSelected 
        data-testid="dropdown-value"
        className='current'
        onClick={() => toggling()}
      >
        {value} <span className='chevron'></span>
      </DropDownSelected>
      {isActive && (
        <DropDownList data-testid="dropdown-list">
          {options.map(item => (
            <DropDownListItem
              key={item}
              onClick={() => changeSelected(item)}
            >
              {item}
            </DropDownListItem>
          ))}
        </DropDownList>
      )}
    </DropDownContainer>
  )
};

const DropDownContainer = styled.div`
  position: relative;
  font-size: 10px;

  @media (min-width: 1100px) {
    font-size: 13px;
  }
`;

const DropDownSelected = styled.div`
  position: relative;
  padding: 6px;
  min-width: 128px;
  border: 2px solid #333333;

  @media (min-width: 1100px) {
    padding: 8px;
  }
`;

const DropDownList = styled.ul`
  position: absolute;
  left: 0;
  right: 0;
  background: #ffffff;
  z-index: 1;
`;

const DropDownListItem = styled.li`
  padding: 6px;
  border-width: 0 2px 2px 2px;
  border-style: solid;
  border-color: #333333;
  cursor: pointer;
`;
