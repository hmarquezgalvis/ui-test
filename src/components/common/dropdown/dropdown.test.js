import { fireEvent, render, screen } from '@testing-library/react';
import { DropDown } from './dropdown';

describe('Tests for <Dropdown />', () => {

  describe('should render and save the snapshot correctly', () => {
    it('default state', () => {
      const { asFragment } = render(<DropDown value="Test 1" />);
      expect(asFragment()).toMatchSnapshot();
    });
    
    it('with some props', () => {
      const { asFragment } = render(<DropDown options={["Test 1"]} value="Test 1" />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('options opened', () => {
      render(<DropDown options={["Test 1"]} value="Test 1" />);
      const button = screen.getByTestId('dropdown-value');
      
      fireEvent(
        button,
        new MouseEvent('click', { bubbles: true, cancelable: true }),
      );
      
      const list = screen.getByTestId('dropdown-list');
      expect(list).toBeDefined();
    });

  });

});