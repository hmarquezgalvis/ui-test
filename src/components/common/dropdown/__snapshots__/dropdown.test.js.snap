// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Dropdown /> should render and save the snapshot correctly default state 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ bGOKaH"
  >
    <div
      class="sc-gsnTZi kSpeQC current"
      data-testid="dropdown-value"
    >
      Test 1 
      <span
        class="chevron"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tests for <Dropdown /> should render and save the snapshot correctly with some props 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ bGOKaH"
  >
    <div
      class="sc-gsnTZi kSpeQC current"
      data-testid="dropdown-value"
    >
      Test 1 
      <span
        class="chevron"
      />
    </div>
  </div>
</DocumentFragment>
`;
